core/chain_makers.go

```diff
@@ -196,14 +197,18 @@ func (b *BlockGen) OffsetTime(seconds int64) {
 // values. Inserting them into BlockChain requires use of FakePow or
 // a similar non-validating proof of work implementation.
 func GenerateChain(config *params.ChainConfig, parent *types.Block, engine consensus.Engine, db ethdb.Database, n int, gen func(int, *BlockGen)) ([]*types.Block, []types.Receipts) {
+	if config.ChainID == nil { // for testing purposes
+		config = params.TestChainConfig
+	}
+
 	if config == nil {
 		config = params.TestChainConfig
 	}
 	blocks, receipts := make(types.Blocks, n), make([]types.Receipts, n)
 	chainreader := &fakeChainReader{config: config}
 	genblock := func(i int, parent *types.Block, statedb *state.StateDB) (*types.Block, types.Receipts) {
 		b := &BlockGen{i: i, chain: blocks, parent: parent, statedb: statedb, config: config, engine: engine}
-		b.header = makeHeader(chainreader, parent, statedb, b.engine)
+		// b.header = makeHeader(chainreader, parent, statedb, b.engine)
```

Comment:
Why not make the headers?

Reply:
Part of GenerateChain, it was probably commented out during debugging before deciding to redo the test entirely.

Reply:
Ok, sounds like this line should be reverted as well.